### ISSUE TITLE
[Edit]: Test & edit Docker tutorial

### ISSUE THREAD

**Issue by hannesdatta** [tilburgsciencehub/website#305]
### Contact Details

h.datta@tilburguniversity.edu

### Describe the problem. Is this a small fix (typos, few sentences...) or does it require more elaborate changes (modifying, adding or removing an entire paragraph or chapter, modifying the content structure...)?

TSH has a new docker tutorial: https://tilburgsciencehub.com/setup/docker

The purpose of this task is to learn Docker, but also test whether our tutorial misses anything.

Outcome:
- learn docker using our tutorial
- try to see whether anything is missing and immediately add it as a PR

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by hannesdatta** [comment]
@srosh2000 , could u accept the git invite and see whether u can try this issue?

**Comment by srosh2000** [comment]
Sure thing!

**Comment by srosh2000** [comment]
- I did a trial run of the tutorial and i could successfully build the image but m stuck when trying to run the docker. 
- Here is the screenshot from my terminal window: 
<img width="575" alt="Screenshot 2022-03-14 at 3 54 46 PM" src="https://user-images.githubusercontent.com/67949747/158198450-153d9071-d842-4121-aecb-61dc89d3e2eb.png">
)

Any leads on how I could debug this? is it related to relative file path or something?

**Comment by hannesdatta** [comment]
it looks like the instruction.sh has not been copied to the machine? best to set up a teamviewer session to check it out if this isn't the thing.

**Comment by hannesdatta** [comment]
@srosh2000, any update?

**Comment by srosh2000** [comment]
<img width="781" alt="Screenshot 2022-03-21 at 5 22 15 PM" src="https://user-images.githubusercontent.com/67949747/159305188-bad4c85d-3aa8-4e58-abd4-3646f9ab0409.png">
I copied all the necessary source code files as shown in the screenshot. So i think thats not the problem? Would be good to set up a teamviewer if this isnt the issue indeed. And sincere apologies for not getting back earlier, I've been bit caught up with some coursework deadlines but will check this up once again.

**Comment by hannesdatta** [comment]
@srosh2000, let's break it down in steps.

Could you please create a new repository on the tilburgsciencehub organization, called `docker-demo`, in which you commit the code that Tomeu has written in that building block? Please add that link to the building block directly.

That way, we offer people to directly get started without having to type code themselves.

In turn, it allows me to directly debug things in that repos. And we're all working in the "same" version.

Makes sense?

**Comment by srosh2000** [comment]
Yes perfect, will do that now.

**Comment by srosh2000** [comment]
Its done! I just created a new repo and added you as collaborator. Hope you can see have access now?

**Comment by hannesdatta** [comment]
@srosh2000 - thanks so much for involving me again on this. I learned SO MUCH!

I have now significantly [updated the repository](https://github.com/tilburgsciencehub/docker-demo), and also reworked the tutorial slightly.

The specific BUG that you reported had to do w/ the fact that the file permissions for (what used to be `instructions.sh`) `run.sh` weren't set properly. I have added a little note on the use of `chmod` in the tutorial. I think I may have permanently fixed this issue by using the `entrypoint` command in the `dockerfile`.

Well - to make things easy: can you redo the tutorial & confirm it runs now?

Thanks a bunch!!!

**Comment by hannesdatta** [comment]
@Tomeulnv - thanks again for this great tutorial. I have just done it myself and found it to work really well. :) Time to spread the word about it I guess. @srosh2000 will confirm she can run it, then we can promote it.

**Comment by srosh2000** [comment]
Some updates from take two of the trial:
- Upon setting the working directory on terminal I ran: chmod +x src/run.sh
- Then i built the image using: docker build -t myname/myimage . --> did some minor debugging of the dockerfile for this step as the working directory has to be set to docker-demo and not docker_rpy anymore based on the restructured version
-  Then I ran the container.. but I still get an error: 
<img width="573" alt="Screenshot 2022-03-23 at 1 12 41 PM" src="https://user-images.githubusercontent.com/67949747/159696386-9392e9ab-7ce9-4109-96a1-bffa384dec83.png">

- Also: I will correct typos in the building block page based on recent changes: like changing docker_rpy to docker-demo everywhere, instructions.sh to run.sh and src instead of code.

**Comment by srosh2000** [comment]
Maybe I'm missing something in my execution?

**Comment by hannesdatta** [comment]
Hi @srosh2000, I looked at your changes again, and committed a new version.

> did some minor debugging of the dockerfile for this step as the working directory has to be set to docker-demo and not docker_rpy anymore based on the restructured version

- The working directory was intentionally left like it - it doesn't need to be called `docker-demo`. 

> Then I ran the container.. but I still get an error:
> `rushini$ docker run-it --rm -v "/Users/roshinisudhararan/Desktop"::/docker-demo" myname/myimage

This is where the mistake happens.

Your working path needs to be in the CURRENT repository, say, the root of where the `src` code folder is located in, and you have to bind it to the working directory.

What you seem to be doing here is binding your DESKTOP to the demo (but it won't see /src on your desktop)!

The best way to try this is using the `$(pwd)` shortcode on your computer. Happy to show you how to do it. It's part of the tutorial btw, but maybe it needs to be written up better.

If you get it fixed on your system, it's probably best if you do a slight rewrite here and there so others don't fall for the same mistake.

Hope it works?

Updated the demo-docker repository btw and found some more issues & fixed them. 

> Also: I will correct typos in the building block page based on recent changes: like changing docker_rpy to docker-demo everywhere, instructions.sh to run.sh and src instead of code.

Please discuss any changes via a PR next - I consider the workflow to be correct for now (tested it like a zillion times). So really think it must be the run command on your system & from where you run it.

**Comment by srosh2000** [comment]
Hi @hannesdatta ,
I certainly think the issues are specific to my laptop. When trying again I started to get some new errors when building the image already this time, so will have a deeper look at it later or show you in our next meeting or something.
But anyhoo good news for now, I tried it on Bilge's laptop and it runs perfectly well! So the workflow is indeed correct. Thanks a lot for this task! It was quite a useful and fun learning experience!  Hope to try and create my own dockerfile soon :D
